pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/model/elastic/filter/query/ActivitiesCollectionQueryBuilderTest.java

```diff
@@ -0,0 +1,108 @@
+package org.broadinstitute.dsm.model.elastic.filter.query;
+
+import org.apache.lucene.search.join.ScoreMode;
+import org.broadinstitute.dsm.model.elastic.filter.FilterParser;
+import org.elasticsearch.index.query.AbstractQueryBuilder;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.ExistsQueryBuilder;
+import org.elasticsearch.index.query.MatchQueryBuilder;
+import org.elasticsearch.index.query.NestedQueryBuilder;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ActivitiesCollectionQueryBuilderTest {
+
+    public static final String ES = "ES";
+    private BaseAbstractQueryBuilder abstractQueryBuilder;
+
+
+    @Test
+    public void activitiesCompletedAtNotEmpty() {
+        String filter = " AND ANGIORELEASE.completedAt IS NOT NULL ";
+        abstractQueryBuilder = AbstractQueryBuilderFactory.create(ES, filter);
+        abstractQueryBuilder.setParser(new FilterParser());
+        AbstractQueryBuilder<?> actual = abstractQueryBuilder.build();
+        BoolQueryBuilder expected = new BoolQueryBuilder();
+        BoolQueryBuilder nestedBoolQuery = new BoolQueryBuilder();
+        nestedBoolQuery.must(new MatchQueryBuilder("activities.activityCode", "ANGIORELEASE"));
+        nestedBoolQuery.must(new BoolQueryBuilder().must(new ExistsQueryBuilder("activities.completedAt")));
+        NestedQueryBuilder expectedNestedQuery = new NestedQueryBuilder("activities", nestedBoolQuery, ScoreMode.Avg);
+        expected.must(expectedNestedQuery);
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void activitiesCreatedAtExactDate() {
+        String filter = " AND ANGIORELEASE.completedAt = '2022-07-28'";
```

Comment:
@gmakharat how much work would it be to add a way to filter for 
` AND ANGIORELEASE.activityVersion = 'v2' `
to only show pts that have a new version of an activity filled out

Reply:
That should not be a big deal I think, probably the best way is to create separate test class for the similar tests but for studies which may have two version.

Reply:
osteo would be a study that has two versions :)